jgrapht-core/src/main/java/org/jgrapht/generate/RandomRegularGraphGenerator.java

```diff
@@ -0,0 +1,234 @@
+
+        // directed/mixed case
+        if (!target.getType().isUndirected()) {
+            throw new IllegalArgumentException("target graph must be undirected");
+        }
+
+        // simple case
+        if (target.getType().isSimple()) {
+            // no nodes or zero degree case
+            if (this.n == 0 || this.d == 0) {
+                EmptyGraphGenerator<V, E> emptyGraphGenerator = new EmptyGraphGenerator<>(this.n);
+                emptyGraphGenerator.generateGraph(target, vertexFactory, resultMap);
+            }
+
+            else if (this.d == this.n) {
+                throw new IllegalArgumentException("target graph must be simple if 'd==n'");
+            }
+
+            // complete case
+            else if (this.d == (this.n - 1)) {
+                CompleteGraphGenerator<V, E> completeGraphGenerator = new CompleteGraphGenerator<>(this.n);
+                completeGraphGenerator.generateGraph(target, vertexFactory, resultMap);
+            }
+
+            // general case
+            else {
+                generateSimpleRegularGraph(target, vertexFactory);
+            }
+        }
+
+        // non-simple case
+        else {
+            generateNonSimpleRegularGraph(target, vertexFactory);
+        }
+    }
+
+
+    private boolean isDRegular(Graph<V, E> target) {
+        for (V v : target.vertexSet()) {
+            if (target.degreeOf(v) != this.d) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+
+    private void generateSimpleRegularGraph(Graph<V, E> target, VertexFactory<V> vertexFactory) {
+        // integers to vertices
+        List<V> vertices = new ArrayList<>();
+        for (int i = 0; i < this.n; i++) {
+            V vertex = vertexFactory.createVertex();
+            vertices.add(vertex);
+            target.addVertex(vertex);
+        }
+
+        while (!isDRegular(target)) {
+            // initialize target graph
+            target.removeAllEdges(new HashSet<>(target.edgeSet()));
+
+            // set of candidate edges
+            Set<Map.Entry<V, V>> S = new HashSet<>();
+            for (int i = 0; i < this.n; i++) {
+                for (int j = i + 1; j < this.n; j++) {
+                    V u = vertices.get(i);
+                    V v = vertices.get(j);
+                    S.add(new AbstractMap.SimpleImmutableEntry<>(u, v));
+                }
+            }
+
+            Set<Map.Entry<V, V>> toRemoveFromS = new HashSet<>();
+            while (!S.isEmpty()) {
+                // needed to normalize probabilities in [0,1]
+                double norm = 0.0;
+                for (Map.Entry<V, V> edge : S) {
+                    V u = edge.getKey();
+                    V v = edge.getValue();
+                    norm += (this.d - target.degreeOf(u)) * (this.d - target.degreeOf(v));
+                }
+
+                double r = Math.random();
```

Comment:
As I mentioned in my general comment, you need to add a private field (Random rng) just like the other generators, and replace this with `rng.nextDouble()`.